### DISCUSSION TITLE
Still having heavy stutters on high-end PC

### DISCUSSION THREAD

**Discussion by minisbett** [ppy/osu#19666] [Q&A] [answered]
### Type

Performance

### Bug description

for over two years now that I've been trying lazer here and there I always had huge performance problems, lazer felt like not even running at 60fps while saying it's running on the most I can get. I got a 240hz monitor and everything fast is just unplayable, on stable the performance is great tho. It's not small lag spikes here and there, it's like having a constant fps cap. I checked Nvidia settings, no fps cap nothing set there.
My specs are a 5600X and a 1660S and if that shouldn't handle lazer then idk.

### Version

2022.731.1

### Logs

[network.log](https://github.com/ppy/osu/files/9289740/network.log)
[performance.log](https://github.com/ppy/osu/files/9289741/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9289742/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9289744/updater.log)
[database.log](https://github.com/ppy/osu/files/9289745/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/9289746/legacy-ipc.log)

**Comment by peppy** [comment, marked as answer]
To confirm, you're saying that the fps that the game shows is high, but it feels low?

If so, there's a few possibilities:
- You're noticing screen tearing. Try switching on vsync
- You're noticing the game not running exclusive fullscreen (adding latency). Check in settings to see whether the game is running exclusive fullscreen, it will show a message if so.

Regardless, the first thing you need to do is diagnose what you're seeing. We can only help if you provide more information, such as a screen capture with the frame graphs showing (Ctrl+F11 twice). If the game is showing perfect frames, then taking a slow motion video of your screen may allow us to help you further.

**Reply by minisbett** [reply]
It doesn't feel like screen tearing and I'm don't think latency would have an impact on the fps itself, only on the time it takes to display but I'll try it. I will also try to do some recording of gameplay

**Reply by minisbett** [reply]
huh, it seems like running it in exclusive fullscreen fixed the issue. I've never seen a game where fullscreen has such a high impact on the fps and I really don't like exclusive fullscreen because everytime you tab out your screen gets black for like 5 seconds but I guess that's the only option here. Thanks a lot for the help, amazing job!

**Reply by peppy** [reply]
The 5 second delay will be reduced once we make the switch of graphical backends, if that's any consolation.
